Automattic/wp-calypso
# GitHub Deployments: support simple repo connect

Related to https://github.com/Automattic/dotcom-forge/issues/5333

See a summary of changes + video here. p1707809411913779-slack-C06D9M3CHMK

<img width="1216" alt="Screenshot 2567-02-12 at 16 20 10" src="https://github.com/Automattic/wp-calypso/assets/6851384/76c64695-0712-4960-a6c9-7f025b4eadb0">

<img width="1239" alt="Screenshot 2567-02-12 at 16 22 26" src="https://github.com/Automattic/wp-calypso/assets/6851384/b98dc2d5-3abd-4dd3-bb96-217d37153028">


## Proposed Changes

This isn't 100% done but can be reviewed now. 

* Support connecting a simple repo
    - List repos for an account where the app is installed
    - Connect to a repo and specify some basic config
* Supports filtering repos by account 
* Support a local filter of repos by name (filters local results only)

Not done: 
- pagination of results. Discussion p1707728079969959-slack-C06D9M3CHMK. 
- connecting to another branch that isn't the default

## Testing Instructions



* Apply patch D137683-code which adds a new endpoint `GET /wpcom/v2/hosting/github/repositories`
* Also make sure you have the access token hardcoded as per previous setup steps
* Start Calypso on this branch and go to `/github-deployments/:atomic-site`
* Select a repo
* Connect the repo and check the network response is 200

## Pre-merge Checklist
